geopandas/tools/overlay.py

```diff
@@ -264,6 +264,20 @@ def overlay(df1, df2, how="intersection", keep_geom_type=None, make_valid=True):
                 "df{} contains mixed geometry types.".format(i + 1)
             )
 
+    box_gdf1 = df1.total_bounds
+    box_gdf2 = df2.total_bounds
+
+    if not (
+        ((box_gdf1[0] <= box_gdf2[2]) and (box_gdf2[0] <= box_gdf1[2]))
+        and ((box_gdf1[1] <= box_gdf2[3]) and (box_gdf2[1] <= box_gdf1[3]))
+    ):
+        return GeoDataFrame(
+            [],
+            columns=df1.drop("geometry", axis=1).columns.to_list()
```

Comment:
My first attempt was to return `df1.join(df2.drop('geometry', axis=1)).loc[:0]` but  I had to do this here in order to pass the test, otherwise the dtypes were not matching. I thought this was better than trying to change the test.